services/backend/src/assignments-controller.test.js

```diff
@@ -0,0 +1,147 @@
+const _ = require('lodash');
+
+const mockDbGet = jest.fn(() => ({}));
+const mockDbScan = jest.fn(() => ({}));
+const mockDbPut = jest.fn(() => ({}));
+const mockDbDelete = jest.fn(() => ({}));
+const jsonMock = jest.fn();
+
+// Mock @logan/aws
+jest.doMock('@logan/aws', () => {
+    const mocked = jest.requireActual('@logan/aws');
+    mocked.dynamoUtils.get = mockDbGet;
+    mocked.dynamoUtils.scan = mockDbScan;
+    mocked.dynamoUtils.put = mockDbPut;
+    mocked.dynamoUtils.delete = mockDbDelete;
+    mocked.secretUtils.getSecret = async () => ({ web: 'mock-secret' });
+    return mocked;
+});
+
+const basicAssignment1 = {
+    aid: 'abc123',
+    uid: 'usr123',
+    title: 'assignment1',
+    cid: 'cid123',
+    description: 'basic assignment 1',
+    dueDate: '1/30/21',
+};
+
+const basicAssignment2 = {
+    aid: 'abc321',
+    uid: 'usr123',
+    title: 'assignment2',
+    cid: 'cid321',
+    description: 'basic assignment 2',
+    dueDate: '1/31/21',
+};
+
+const assignmentsController = require('./assignments-controller');
+
+beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+});
+
+describe('getAssignment', () => {
+    it('Basic fetch returns the correct assignment', async () => {
+        const req = {
+            params: { aid: 'abc123' },
+        };
+        const dbFormat = {
+            aid: 'abc123',
+            uid: 'usr123',
+            title: 'assignment1',
+            cid: 'cid123',
+            desc: 'basic assignment 1',
+            due: '1/30/21',
+        };
```

Comment:
Instead of doing this, you can export `toDbFormat` and `fromDbFormat` and use them here for more convenient parsing. I'll add suggestions to explain

Reply:
Your module.exports for the assignments controller can look like this:
```js
module.exports = {
    __test_only__: { toDbFormat, fromDbFormat },
    getAssignment,
    getAssignments,
    createAssignment,
    updateAssignment,
    deleteAssignment,
};
```

and then when you import the assignments controller here, you can do:

```js
// eslint-disable-next-line import/newline-after-import
const assignmentsController = require('./assignments-controller');
const { toDbFormat, fromDbFormat } = assignmentsController.__test_only__;
```

and you can use those functions to format appropriately